DatadogTrace/Sources/DatadogTracer.swift

```diff
@@ -8,7 +8,8 @@ import Foundation
 import DatadogInternal
 
 internal class DatadogTracer: OTTracer {
-    internal weak var core: DatadogCoreProtocol?
+    /// Trace feature scope.
+    private let featureScope: FeatureScope
```

Comment:
💡 to achieve mid-term goals ("no core dependency") from https://github.com/DataDog/dd-sdk-ios/pull/1744. With this change we facilitate testing `DatadogTracer` as part of `DatadogTraceTests` using single thread (vs asynchronous testing with core mock in `DatadogCoreTests`).